ucsmsdk_samples/firmware/ucsfirmware.py

```diff
@@ -388,6 +398,11 @@ def firmware_add_remote(handle, file_name, remote_path, protocol, server,
     handle.add_mo(firmware_downloader)
     # handle.set_dump_xml()
     handle.commit()
+    
+    time.sleep(5)
+    if return_download_state(handle, file_name) == 'failed':
```

Comment:
Better poll for `TRANSFER_STATE_DOWNLOADED` and raise exception if `TRANSFER_STATE_FAILED`. 
@vvb - should we add `timeout` here?